### ISSUE TITLE
Updating reaches takes very long

### ISSUE THREAD

**Issue by max-trolliet** [QGEP/QGEP#437]
Changing the geometry (or attributes) of a reach takes 1 minute.

Adding some debug prints to on update triggers on various objects reveals that it's traversing a lot (all?) of the network and recalculating style/labels.

Also happens when triggering the change with a simple `UPDATE` directly on the DB.

Database version 1.1.0

The explain:

```
"Update on reach_point  (cost=10.00..45.57 rows=1 width=380)"
"  ->  Nested Loop Left Join  (cost=10.00..45.57 rows=1 width=380)"
"        ->  Nested Loop Left Join  (cost=9.72..45.20 rows=1 width=391)"
"              ->  Nested Loop Left Join  (cost=9.44..36.89 rows=1 width=385)"
"                    ->  Nested Loop Left Join  (cost=9.15..28.58 rows=1 width=396)"
"                          Join Filter: ((ne.obj_id)::text = (re.obj_id)::text)"
"                          ->  Nested Loop  (cost=8.88..20.27 rows=1 width=390)"
"                                ->  Nested Loop  (cost=8.59..19.88 rows=1 width=401)"
"                                      ->  Hash Right Join  (cost=8.31..11.57 rows=1 width=178)"
"                                            Hash Cond: ((pp.obj_id)::text = (re.fk_pipe_profile)::text)"
"                                            ->  Seq Scan on pipe_profile pp  (cost=0.00..2.91 rows=91 width=23)"
"                                            ->  Hash  (cost=8.30..8.30 rows=1 width=222)"
"                                                  ->  Index Scan using pkey_qgep_od_reach_obj_id on reach re  (cost=0.28..8.30 rows=1 width=222)"
"                                                        Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                      ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_from  (cost=0.28..8.30 rows=1 width=223)"
"                                            Index Cond: ((obj_id)::text = (re.fk_reach_point_from)::text)"
"                                ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point  (cost=0.28..0.38 rows=1 width=23)"
"                                      Index Cond: ((obj_id)::text = (rp_from.obj_id)::text)"
"                          ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement ne  (cost=0.28..8.30 rows=1 width=40)"
"                                Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                    ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_to  (cost=0.28..8.30 rows=1 width=23)"
"                          Index Cond: ((obj_id)::text = (re.fk_reach_point_to)::text)"
"              ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure ws  (cost=0.28..8.30 rows=1 width=23)"
"                    Index Cond: ((ne.fk_wastewater_structure)::text = (obj_id)::text)"
"        ->  Index Scan using pkey_qgep_od_channel_obj_id on channel ch  (cost=0.28..0.34 rows=1 width=23)"
"              Index Cond: ((obj_id)::text = (ws.obj_id)::text)"
""
"Update on reach_point  (cost=10.00..45.57 rows=1 width=380)"
"  ->  Nested Loop Left Join  (cost=10.00..45.57 rows=1 width=380)"
"        ->  Nested Loop Left Join  (cost=9.72..45.20 rows=1 width=391)"
"              ->  Nested Loop Left Join  (cost=9.44..36.89 rows=1 width=385)"
"                    ->  Nested Loop Left Join  (cost=9.15..28.58 rows=1 width=396)"
"                          Join Filter: ((ne.obj_id)::text = (re.obj_id)::text)"
"                          ->  Nested Loop  (cost=8.88..20.27 rows=1 width=390)"
"                                ->  Nested Loop  (cost=8.59..19.88 rows=1 width=401)"
"                                      ->  Hash Right Join  (cost=8.31..11.57 rows=1 width=178)"
"                                            Hash Cond: ((pp.obj_id)::text = (re.fk_pipe_profile)::text)"
"                                            ->  Seq Scan on pipe_profile pp  (cost=0.00..2.91 rows=91 width=23)"
"                                            ->  Hash  (cost=8.30..8.30 rows=1 width=222)"
"                                                  ->  Index Scan using pkey_qgep_od_reach_obj_id on reach re  (cost=0.28..8.30 rows=1 width=222)"
"                                                        Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                      ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_to  (cost=0.28..8.30 rows=1 width=223)"
"                                            Index Cond: ((obj_id)::text = (re.fk_reach_point_to)::text)"
"                                ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point  (cost=0.28..0.38 rows=1 width=23)"
"                                      Index Cond: ((obj_id)::text = (rp_to.obj_id)::text)"
"                          ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement ne  (cost=0.28..8.30 rows=1 width=40)"
"                                Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                    ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_from  (cost=0.28..8.30 rows=1 width=23)"
"                          Index Cond: ((obj_id)::text = (re.fk_reach_point_from)::text)"
"              ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure ws  (cost=0.28..8.30 rows=1 width=23)"
"                    Index Cond: ((ne.fk_wastewater_structure)::text = (obj_id)::text)"
"        ->  Index Scan using pkey_qgep_od_channel_obj_id on channel ch  (cost=0.28..0.34 rows=1 width=23)"
"              Index Cond: ((obj_id)::text = (ws.obj_id)::text)"
""
"Update on channel  (cost=9.99..53.47 rows=1 width=115)"
"  ->  Nested Loop Left Join  (cost=9.99..53.47 rows=1 width=115)"
"        ->  Nested Loop Left Join  (cost=9.71..53.11 rows=1 width=76)"
"              ->  Nested Loop Left Join  (cost=9.43..44.81 rows=1 width=70)"
"                    ->  Nested Loop Left Join  (cost=9.15..36.50 rows=1 width=81)"
"                          ->  Nested Loop  (cost=8.87..28.19 rows=1 width=92)"
"                                ->  Nested Loop  (cost=8.59..19.88 rows=1 width=69)"
"                                      ->  Hash Right Join  (cost=8.31..11.57 rows=1 width=63)"
"                                            Hash Cond: ((pp.obj_id)::text = (re.fk_pipe_profile)::text)"
"                                            ->  Seq Scan on pipe_profile pp  (cost=0.00..2.91 rows=91 width=23)"
"                                            ->  Hash  (cost=8.30..8.30 rows=1 width=107)"
"                                                  ->  Index Scan using pkey_qgep_od_reach_obj_id on reach re  (cost=0.28..8.30 rows=1 width=107)"
"                                                        Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                      ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement ne  (cost=0.28..8.30 rows=1 width=40)"
"                                            Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                ->  Index Scan using pkey_qgep_od_channel_obj_id on channel  (cost=0.28..8.30 rows=1 width=23)"
"                                      Index Cond: ((obj_id)::text = (ne.fk_wastewater_structure)::text)"
"                          ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_from  (cost=0.28..8.30 rows=1 width=23)"
"                                Index Cond: ((obj_id)::text = (re.fk_reach_point_from)::text)"
"                    ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_to  (cost=0.28..8.30 rows=1 width=23)"
"                          Index Cond: ((obj_id)::text = (re.fk_reach_point_to)::text)"
"              ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure ws  (cost=0.28..8.30 rows=1 width=23)"
"                    Index Cond: ((ne.fk_wastewater_structure)::text = (obj_id)::text)"
"        ->  Index Scan using pkey_qgep_od_channel_obj_id on channel ch  (cost=0.28..0.34 rows=1 width=73)"
"              Index Cond: ((obj_id)::text = (ws.obj_id)::text)"
""
"Update on wastewater_structure  (cost=9.99..53.47 rows=1 width=1156)"
"  ->  Nested Loop Left Join  (cost=9.99..53.47 rows=1 width=1156)"
"        ->  Nested Loop Left Join  (cost=9.71..53.11 rows=1 width=1167)"
"              ->  Nested Loop Left Join  (cost=9.43..44.81 rows=1 width=184)"
"                    ->  Nested Loop Left Join  (cost=9.15..36.50 rows=1 width=195)"
"                          ->  Nested Loop  (cost=8.87..28.19 rows=1 width=206)"
"                                ->  Nested Loop  (cost=8.59..19.88 rows=1 width=111)"
"                                      ->  Hash Right Join  (cost=8.31..11.57 rows=1 width=63)"
"                                            Hash Cond: ((pp.obj_id)::text = (re.fk_pipe_profile)::text)"
"                                            ->  Seq Scan on pipe_profile pp  (cost=0.00..2.91 rows=91 width=23)"
"                                            ->  Hash  (cost=8.30..8.30 rows=1 width=107)"
"                                                  ->  Index Scan using pkey_qgep_od_reach_obj_id on reach re  (cost=0.28..8.30 rows=1 width=107)"
"                                                        Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                      ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement ne  (cost=0.28..8.30 rows=1 width=82)"
"                                            Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure  (cost=0.28..8.30 rows=1 width=95)"
"                                      Index Cond: ((obj_id)::text = (ne.fk_wastewater_structure)::text)"
"                          ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_from  (cost=0.28..8.30 rows=1 width=23)"
"                                Index Cond: ((obj_id)::text = (re.fk_reach_point_from)::text)"
"                    ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_to  (cost=0.28..8.30 rows=1 width=23)"
"                          Index Cond: ((obj_id)::text = (re.fk_reach_point_to)::text)"
"              ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure ws  (cost=0.28..8.30 rows=1 width=1000)"
"                    Index Cond: ((ne.fk_wastewater_structure)::text = (obj_id)::text)"
"        ->  Index Scan using pkey_qgep_od_channel_obj_id on channel ch  (cost=0.28..0.34 rows=1 width=23)"
"              Index Cond: ((obj_id)::text = (ws.obj_id)::text)"
""
"Update on wastewater_networkelement  (cost=9.99..53.47 rows=1 width=140)"
"  ->  Nested Loop Left Join  (cost=9.99..53.47 rows=1 width=140)"
"        ->  Nested Loop Left Join  (cost=9.71..53.12 rows=1 width=151)"
"              ->  Nested Loop Left Join  (cost=9.43..44.81 rows=1 width=128)"
"                    ->  Nested Loop Left Join  (cost=9.15..36.50 rows=1 width=139)"
"                          ->  Nested Loop Left Join  (cost=8.87..28.19 rows=1 width=150)"
"                                Join Filter: ((ne.obj_id)::text = (re.obj_id)::text)"
"                                ->  Nested Loop  (cost=8.59..19.88 rows=1 width=86)"
"                                      ->  Hash Right Join  (cost=8.31..11.57 rows=1 width=63)"
"                                            Hash Cond: ((pp.obj_id)::text = (re.fk_pipe_profile)::text)"
"                                            ->  Seq Scan on pipe_profile pp  (cost=0.00..2.91 rows=91 width=23)"
"                                            ->  Hash  (cost=8.30..8.30 rows=1 width=107)"
"                                                  ->  Index Scan using pkey_qgep_od_reach_obj_id on reach re  (cost=0.28..8.30 rows=1 width=107)"
"                                                        Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                      ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement  (cost=0.28..8.30 rows=1 width=23)"
"                                            Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement ne  (cost=0.28..8.30 rows=1 width=98)"
"                                      Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                          ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_from  (cost=0.28..8.30 rows=1 width=23)"
"                                Index Cond: ((obj_id)::text = (re.fk_reach_point_from)::text)"
"                    ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_to  (cost=0.28..8.30 rows=1 width=23)"
"                          Index Cond: ((obj_id)::text = (re.fk_reach_point_to)::text)"
"              ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure ws  (cost=0.28..8.30 rows=1 width=23)"
"                    Index Cond: ((ne.fk_wastewater_structure)::text = (obj_id)::text)"
"        ->  Index Scan using pkey_qgep_od_channel_obj_id on channel ch  (cost=0.28..0.34 rows=1 width=23)"
"              Index Cond: ((obj_id)::text = (ws.obj_id)::text)"
""
"Update on reach  (cost=9.99..53.48 rows=1 width=324)"
"  ->  Nested Loop Left Join  (cost=9.99..53.48 rows=1 width=324)"
"        ->  Nested Loop Left Join  (cost=9.71..53.12 rows=1 width=335)"
"              ->  Nested Loop Left Join  (cost=9.43..44.81 rows=1 width=329)"
"                    ->  Nested Loop Left Join  (cost=9.15..36.50 rows=1 width=340)"
"                          ->  Nested Loop Left Join  (cost=8.87..28.19 rows=1 width=351)"
"                                Join Filter: ((ne.obj_id)::text = (re.obj_id)::text)"
"                                ->  Nested Loop  (cost=8.59..19.88 rows=1 width=345)"
"                                      ->  Hash Right Join  (cost=8.31..11.57 rows=1 width=288)"
"                                            Hash Cond: ((pp.obj_id)::text = (re.fk_pipe_profile)::text)"
"                                            ->  Seq Scan on pipe_profile pp  (cost=0.00..2.91 rows=91 width=23)"
"                                            ->  Hash  (cost=8.30..8.30 rows=1 width=282)"
"                                                  ->  Index Scan using pkey_qgep_od_reach_obj_id on reach re  (cost=0.28..8.30 rows=1 width=282)"
"                                                        Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                      ->  Index Scan using pkey_qgep_od_reach_obj_id on reach  (cost=0.28..8.30 rows=1 width=57)"
"                                            Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                                ->  Index Scan using pkey_qgep_od_wastewater_networkelement_obj_id on wastewater_networkelement ne  (cost=0.28..8.30 rows=1 width=40)"
"                                      Index Cond: ((obj_id)::text = 'ch17f516RE002872'::text)"
"                          ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_from  (cost=0.28..8.30 rows=1 width=23)"
"                                Index Cond: ((obj_id)::text = (re.fk_reach_point_from)::text)"
"                    ->  Index Scan using pkey_qgep_od_reach_point_obj_id on reach_point rp_to  (cost=0.28..8.30 rows=1 width=23)"
"                          Index Cond: ((obj_id)::text = (re.fk_reach_point_to)::text)"
"              ->  Index Scan using pkey_qgep_od_wastewater_structure_obj_id on wastewater_structure ws  (cost=0.28..8.30 rows=1 width=23)"
"                    Index Cond: ((ne.fk_wastewater_structure)::text = (obj_id)::text)"
"        ->  Index Scan using pkey_qgep_od_channel_obj_id on channel ch  (cost=0.28..0.34 rows=1 width=23)"
"              Index Cond: ((obj_id)::text = (ws.obj_id)::text)"
```

**Comment by max-trolliet** [comment]
Is fixed by https://github.com/QGEP/datamodel/pull/77 already.

What's missing for a new release that comes with this fix?

**Comment by urskaufmann** [comment]
I think is fixed since datamodel 1.1.1
